<Type Name="Catch&lt;TException&gt;" FullName="System.Activities.Statements.Catch&lt;TException&gt;">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="0f1b6c071e42c7a1058579fd3fca964ffad41b85" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="ko-KR" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
  </Metadata>
  <TypeSignature Language="C#" Value="public sealed class Catch&lt;TException&gt; : System.Activities.Statements.Catch where TException : Exception" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed beforefieldinit Catch`1&lt;(class System.Exception) TException&gt; extends System.Activities.Statements.Catch" />
  <TypeSignature Language="DocId" Value="T:System.Activities.Statements.Catch`1" />
  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class Catch(Of TException)&#xA;Inherits Catch" />
  <TypeSignature Language="C++ CLI" Value="generic &lt;typename TException&gt;&#xA; where TException : Exceptionpublic ref class Catch sealed : System::Activities::Statements::Catch" />
  <AssemblyInfo>
    <AssemblyName>System.Activities</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <TypeParameters>
    <TypeParameter Name="TException">
      <Constraints>
        <BaseTypeName>System.Exception</BaseTypeName>
      </Constraints>
    </TypeParameter>
  </TypeParameters>
  <Base>
    <BaseTypeName>System.Activities.Statements.Catch</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Windows.Markup.ContentProperty("Action")</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <typeparam name="TException">
      <span data-ttu-id="4fbfc-101">
        <see cref="T:System.Activities.Statements.Catch`1" /> 블록에서 처리 중인 예외의 형식입니다.</span>
      <span class="sxs-lookup">
        <span data-stu-id="4fbfc-101">The type of exception that the <see cref="T:System.Activities.Statements.Catch`1" /> block is handling.</span>
      </span>
    </typeparam>
    <summary>
      <span data-ttu-id="4fbfc-102">
        <see cref="T:System.Activities.Statements.TryCatchFinally" /> 활동의 해당 Try 블록에서 예외가 발생한 후 실행할 동작을 포함합니다.</span>
      <span class="sxs-lookup">
        <span data-stu-id="4fbfc-102">Contains the actions to be executed after an exception is raised in the corresponding Try block of a <see cref="T:System.Activities.Statements.TryCatchFinally" /> activity.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4fbfc-103">디자이너에서 한 번 배치 <xref:System.Activities.Statements.Catch> 활동 디자이너에서 다시 정렬할 수 없습니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-103">Once placed in the designer, <xref:System.Activities.Statements.Catch> activities cannot be reordered in the designer.</span></span>  <span data-ttu-id="4fbfc-104">컬렉션의 순서를 변경 하려면 <xref:System.Activities.Statements.Catch> 활동을 삭제 하 고 올바른 순서로 다시 추가할 하거나 워크플로에 대 한 xaml (코드) 뷰를 사용 합니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-104">To reorder the collection of <xref:System.Activities.Statements.Catch> activities, either delete and re-add them in the correct order, or use the xaml (code) view for the workflow.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="4fbfc-105">다음 코드 샘플에서는 <xref:System.Activities.Statements.Catch%601> 활동을 만드는 방법을 보여 줍니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-105">The following code sample demonstrates creating a <xref:System.Activities.Statements.Catch%601> activity.</span></span> <span data-ttu-id="4fbfc-106">이 예제에서는 [순서도 활동 사용 하 여 TryCatch에서 오류 처리](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) 샘플.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-106">This example is from the [Fault Handling in a Flowchart Activity Using TryCatch](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) sample.</span></span>  
  
 [!code-csharp[wfs_FlowChartWithFaultHandling#1](~/samples/snippets/csharp/VS_Snippets_CFX/wfs_flowchartwithfaulthandling/cs/program.cs#1)]  
  
 ]]></format>
    </remarks>
  </Docs>
  <Members>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public Catch ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Activities.Statements.Catch`1.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; Catch();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="4fbfc-107">
            <see cref="T:System.Activities.Statements.Catch\`1" /> 클래스의 새 인스턴스를 만듭니다.</span>
          <span class="sxs-lookup">
            <span data-stu-id="4fbfc-107">Creates a new instance of the <see cref="T:System.Activities.Statements.Catch`1" /> class.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Examples  
 <span data-ttu-id="4fbfc-108">다음 코드 샘플에서는 <xref:System.Activities.Statements.Catch%601> 활동을 만드는 방법을 보여 줍니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-108">The following code sample demonstrates creating a <xref:System.Activities.Statements.Catch%601> activity.</span></span> <span data-ttu-id="4fbfc-109">이 예제에서는 [순서도 활동 사용 하 여 TryCatch에서 오류 처리](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) 샘플.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-109">This example is from the [Fault Handling in a Flowchart Activity Using TryCatch](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) sample.</span></span>  
  
 [!code-csharp[wfs_FlowChartWithFaultHandling#1](~/samples/snippets/csharp/VS_Snippets_CFX/wfs_flowchartwithfaulthandling/cs/program.cs#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="Action">
      <MemberSignature Language="C#" Value="public System.Activities.ActivityAction&lt;TException&gt; Action { get; set; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Activities.ActivityAction`1&lt;!TException&gt; Action" />
      <MemberSignature Language="DocId" Value="P:System.Activities.Statements.Catch`1.Action" />
      <MemberSignature Language="VB.NET" Value="Public Property Action As ActivityAction(Of TException)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; property System::Activities::ActivityAction&lt;TException&gt; ^ Action { System::Activities::ActivityAction&lt;TException&gt; ^ get(); void set(System::Activities::ActivityAction&lt;TException&gt; ^ value); };" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute>
          <AttributeName>System.ComponentModel.DefaultValue(null)</AttributeName>
        </Attribute>
      </Attributes>
      <ReturnValue>
        <ReturnType>System.Activities.ActivityAction&lt;TException&gt;</ReturnType>
      </ReturnValue>
      <Docs>
        <summary>
          <span data-ttu-id="4fbfc-110">catch되는 예외의 처리기입니다.</span>
          <span class="sxs-lookup">
            <span data-stu-id="4fbfc-110">The handler for the exception being caught.</span>
          </span>
        </summary>
        <value>
          <span data-ttu-id="4fbfc-111">작업 동작입니다.</span>
          <span class="sxs-lookup">
            <span data-stu-id="4fbfc-111">The activity action.</span>
          </span>
        </value>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="4fbfc-112">예외 처리기는 <xref:System.Activities.ActivityAction%601> 개체로 표현됩니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-112">The exception handler is represented by a <xref:System.Activities.ActivityAction%601> object.</span></span>  
  
   
  
## Examples  
 <span data-ttu-id="4fbfc-113">다음 코드 샘플에서는 <xref:System.Activities.Statements.Catch%601> 활동의 Action 속성을 할당하는 방법을 보여 줍니다.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-113">The following code sample demonstrates assigning the Action property of a <xref:System.Activities.Statements.Catch%601> activity.</span></span> <span data-ttu-id="4fbfc-114">이 예제에서는 [순서도 활동 사용 하 여 TryCatch에서 오류 처리](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) 샘플.</span><span class="sxs-lookup"><span data-stu-id="4fbfc-114">This example is from the [Fault Handling in a Flowchart Activity Using TryCatch](~/docs/framework/windows-workflow-foundation/samples/fault-handling-in-a-flowchart-activity-using-trycatch.md) sample.</span></span>  
  
 [!code-csharp[wfs_FlowChartWithFaultHandling#1](~/samples/snippets/csharp/VS_Snippets_CFX/wfs_flowchartwithfaulthandling/cs/program.cs#1)]  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName="ExceptionType">
      <MemberSignature Language="C#" Value="public override Type ExceptionType { get; }" />
      <MemberSignature Language="ILAsm" Value=".property instance class System.Type ExceptionType" />
      <MemberSignature Language="DocId" Value="P:System.Activities.Statements.Catch`1.ExceptionType" />
      <MemberSignature Language="VB.NET" Value="Public Overrides ReadOnly Property ExceptionType As Type" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual property Type ^ ExceptionType { Type ^ get(); };" />
      <MemberType>Property</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Activities</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Type</ReturnType>
      </ReturnValue>
      <Docs>
        <summary>
          <span data-ttu-id="4fbfc-115">
            <see cref="T:System.Activities.Statements.Catch" /> 블록에서 처리 중인 예외의 형식입니다.</span>
          <span class="sxs-lookup">
            <span data-stu-id="4fbfc-115">The type of exception that the <see cref="T:System.Activities.Statements.Catch" /> block is handling.</span>
          </span>
        </summary>
        <value>
          <span data-ttu-id="4fbfc-116">예외 형식입니다.</span>
          <span class="sxs-lookup">
            <span data-stu-id="4fbfc-116">The exception type.</span>
          </span>
        </value>
        <remarks>To be added.</remarks>
      </Docs>
    </Member>
  </Members>
</Type>